android/java/org/chromium/chrome/browser/brave_news/CardBuilderFeedCard.java

```diff
@@ -0,0 +1,1421 @@
+                                adTitle.setEllipsize(TextUtils.TruncateAt.END);
+                                adTitle.setLayoutParams(adItemsParams);
+
+                                adItemsParams = new TableRow.LayoutParams(TableRow.LayoutParams.WRAP_CONTENT,
+                                    TableRow.LayoutParams.MATCH_PARENT);
+                                adItemsParams.gravity = Gravity.CENTER_VERTICAL;
+                                adDesc.setGravity(Gravity.CENTER_VERTICAL);
+                                adItemsParams.weight = 1;
+                                adItemsParams.width = 0;
+                                adDesc.setLayoutParams(adItemsParams);
+                                adTitle.setText(adData.title);
+
+                                adDesc.setTextColor(mActivity.getResources().getColor(R.color.news_time_color));
+                                adDesc.setTextSize(12);
+                                adDesc.setText(adData.description);
+
+                                adButtonParams.width = TableRow.LayoutParams.WRAP_CONTENT; 
+                                adButtonParams.height = 80;//TableRow.LayoutParams.WRAP_CONTENT; 
+                                adButton.setPadding(30, 0, 30, 0);
+                                adButton.setTextSize(13);
+                                adButton.setAllCaps(false);
+                                GradientDrawable adButtonBG = new GradientDrawable();
+                                adButtonBG.setColor(mActivity.getResources().getColor(android.R.color.transparent));
+                                adButtonBG.setCornerRadius(55);
+                                adButtonBG.setStroke(1, mActivity.getResources().getColor(R.color.news_text_color));
+                                adButton.setBackground(adButtonBG);
+                                adButton.setText(adData.ctaText);
+                                adButton.setTextColor(mActivity.getResources().getColor(R.color.news_text_color));
+                                adButton.setLayoutParams(adButtonParams);
+                                adButton.setOnClickListener(new View.OnClickListener() {
+                                    @Override
+                                    public void onClick(View v) {
+                                        mBraveNewsController.onDisplayAdVisit(adData.uuid, adData.creativeInstanceId);
+                                        BraveActivity.getBraveActivity().setNewsFeedScrollPosition(position);
+                                        TabUtils.openUrlInSameTab(adData.targetUrl.url);
+                                    }
+                                });
+                            }
+                        });
+                    
+                    } catch (Exception e){
+                        Log.e("bn", "displayad Exception" + e.getMessage());
```

Comment:
this class has `private String TAG = "BN";`

Reply:
replaced